tests/integration/trace/test_zipkin_trace.py

```diff
@@ -136,3 +137,44 @@ def test_tcollector_submit(trace_server):
     results = yield TChannelZipkinTracer(tchannel).record([(trace, anns)])
 
     assert results[0].body.ok is True
+
+
+@pytest.mark.gen_test
+def test_tcollector_submit_never_retry():
+
+    def submit(*args, **kwargs):
+        count[0] += 1
+        if f.running():
+            f.set_result(None)
+        raise Exception()
+
+    count = [0]
+    f = Future()
+
+    zipkin_server = TChannel('zipkin')
+    zipkin_server.thrift.register(TCollector, handler=submit)
+    zipkin_server.listen()
+
+    zipkin_server1 = TChannel('server')
+    zipkin_server1.thrift.register(TCollector, handler=submit)
+
+    @zipkin_server1.raw.register
+    def hello(request):
+        return 'hello'
```

Comment:
Not a zipkin sever. That implies this implements the tcollector interface. Please call it something else.